scrapy_poet/downloadermiddlewares.py

```diff
@@ -91,6 +93,34 @@ def process_request(
         self.crawler.stats.inc_value("scrapy_poet/dummy_response_count")
         return DummyResponse(url=request.url, request=request)
 
+    def _skip_dependency_creation(
+        self, request: Request, response: Response, spider: Spider
+    ) -> bool:
+        """See: https://github.com/scrapinghub/scrapy-poet/issues/48"""
+
+        # No need to skip if the callback doesn't default to the parse() method
+        if request.callback is not None:
+            return False
+
+        # If the Request.cb_kwargs possess all of the cb dependencies, then no
+        # warning message should be issued.
```

Comment:
I just realized while writing this part that the `Injector` doesn't take into account any of the `request.cb_kwargs`. This means it's building it all of the dependencies which could be replaced later on by any `request.cb_kwargs` that are present.

Reply:
Do you see any issue with that?

Reply:
There could be some use cases where building a particular dependency might be too expensive and one approach to avoid such a cost is to provide it manually via `cb_kwargs`.

It also removes the user's escape hatch of avoiding the providers from being used on certain requests since they can provide it themselves within the spider's request.

We could address it later in another PR.